src/api/routes/address.ts

```diff
@@ -277,27 +393,62 @@ export function createAddressRouter(db: PgStore, chainId: ChainID): express.Rout
         }
       });
       if (result) {
-        setETagCacheHeaders(res);
-        res.json(result);
+        await reply.send(result);
       } else {
-        res.status(404).json({ error: 'No matching transaction found' });
+        throw new NotFoundError(`No matching transaction found`);
       }
-    })
+    }
   );
 
   /**
    * @deprecated See `/v2/addresses/:address/transactions`
    */
-  router.get(
-    '/:stx_address/transactions_with_transfers',
-    cacheHandler,
-    asyncHandler(async (req, res, next) => {
-      const stxAddress = req.params['stx_address'];
+  fastify.get(
+    '/:principal/transactions_with_transfers',
```

Comment:
Question: should we take this opportunity to get rid of this one for good? (Not sure if this refactor will be a breaking change or not)